File: problems/algorithms/string-util/js/response/string-util.js
/**
 * text: string
 * action: lowercase, uppercase, camelcase, snakecase, reverse,
 *         countchar, countword, countline
 */
function formatter(text, action) {
  switch (action) {
    case 'lowercase':
      return text.toLowerCase();
    case 'uppercase':
      return text.toUpperCase();
    case 'camelcase':
      return toCamelCase(text);
    case 'snakecase':
      return text.replace(/\s/g, '_');
    case 'reverse':
      return text
        .split('')
        .reverse()
        .join('');
    case 'countchar':
      return text.split('').length;
    case 'countword':
      return text.split(/\s/g).length;
    case 'countline':
      return text.split('\n').length;
  }
}

function toCamelCase(text) {
  let result = [];
  for (let word of text.split(' ')) {
    let firstLetter = word[0].toUpperCase();
    let slicedWord = word.substr(1).toLowerCase();
    result.push(firstLetter + slicedWord);
  }
  return result.join(' ');
}

export { formatter };

File: src/pages/exercises/string-util/codes/ruby/response/src/string_util.rb
# text: string
# action: lowercase, uppercase, camelcase, snakecase, reverse,
#         countchar, countword, countline
#
def formatter(text, action)
  case action
    when 'lowercase'
      return text.downcase
    when 'uppercase'
      return text.upcase
    when 'camelcase'
      return text.downcase.gsub(/\b\w/){$&.upcase}
    when 'snakecase'
      return text.gsub! ' ', '_'
    when 'reverse'
      return text
        .split('')
        .reverse()
        .join('')
    when 'countchar'
      return text.gsub(' ', '').split('').length
    when 'countword'
      return text.gsub('\n',' ').split(' ').length
    when 'countline'
      return text.split('\n').length
  end
end

